File: LAB/Assignment_3/Selection_Sort/Selection_sort.cpp
#include<bits/stdc++.h>
using namespace std;


void selection_sort(vector<int>&arr){
    int n = arr.size();
    for(int i=0; i<n-1; i++){
        int min_val = i;
        for(int j=i+1; j<n; j++){
            if(arr[j] < arr[min_val]){
                min_val = j;
            }
        }
        swap(arr[i], arr[min_val]);
    }
}


void print_array(vector<int> arr){
    for(int i=0; i<arr.size(); i++){
        cout<<arr[i]<<" ";
    }
    cout<<"\n";
}

int main(){

    vector<int> arr = {64, 25, 12, 22, 11};
    cout<<"Original Array: "<<"\n";
    print_array(arr);

    selection_sort(arr);
    cout<<"Sorted Array"<<"\n";
    print_array(arr);


    return 0;
}

File: LAB/Assignment_3/Selection_Sort/Selection_sort.java
import java.util.Arrays;

class Selection_sort{
    static void Selection_sort(int [] arr){
        int n = arr.length;
        for(int i=0; i<n; i++){
            int min_val = i;

            for(int j= i+ 1; j<n; j++){
                if(arr[j] < arr[min_val]){
                    min_val = j;
                }
            }
            int temp = arr[i];
            arr[i] = arr[min_val];
            arr[min_val] = temp;
        }

    } 

    static void print_array(int [] arr){
        for(int val:arr){
            System.out.println(val + " ");
        }
        System.out.println();
    }

    public static void main (String[] args){
        int [] arg = {25,89,17,57,69};
        System.out.println("Original Array: ");
        print_array(arg);
        System.out.print("Sorted Array: ");
        Selection_sort(arg);
        print_array(arg);
    }
}